### PR TITLE
 Add new variable S3_ENDPOINT_URL

### PR DESCRIPTION
**Pull request by stroitelev**
Hello, I had a task to deploy the S3 exporter and integrate with our local S3 service.
I have added new variable for that.
Thank you for your work and maybe my little update will be helpful for you too.

### CODE REVIEW

**Comment by ribbybibby** [comment]
Thanks for the PR @stroitelev! 

Could you please implement this variable as a kingpin flag, rather than reading the environment variable with `os.Getenv()`? A kingpin flag can be provided as an environment variable as well as a commandline flag so it provides more flexibility and also keeps configuration unified to kingpin. 

https://godoc.org/github.com/alecthomas/kingpin#ArgClause.Envar
https://godoc.org/github.com/alecthomas/kingpin#Application.DefaultEnvars

**Comment by stroitelev** [comment]
Hello @ribbybibby! 

Sorry for delay, I have fixed it, could you check please? And sorry for my English, feel free to rewrite my descriptions if something wrong.

P.S. I also have helm chart for the s3-exporter, if it useful I can add it too.

File: s3_exporter.go [hunk 1]
```diff
@@ -144,6 +145,7 @@ func main() {
 		listenAddress = kingpin.Flag("web.listen-address", "Address to listen on for web interface and telemetry.").Default(":9340").String()
 		metricsPath   = kingpin.Flag("web.metrics-path", "Path under which to expose metrics").Default("/metrics").String()
 		probePath     = kingpin.Flag("web.probe-path", "Path under which to expose the probe endpoint").Default("/probe").String()
+		endpointUrl   = kingpin.Flag("s3.endpoint-url", "Custom endpoint URL for S3 service").Default("").Envar("S3_ENDPOINT_URL").String()
```
Comment (ribbybibby) [inline s3_exporter.go]: Could you change this variable name to `endpointURL`, in line with golangs standard for [initialisms](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)?

**Review by ribbybibby — CHANGES_REQUESTED** [review]
Hm, now that I think about it, it occurs to me that if we're exporting this flag as an environment variable we might as well export all of them with `.DefaultEnvars()`:
```patch
diff --git a/s3_exporter.go b/s3_exporter.go
index 08b7c72..535a889 100644
--- a/s3_exporter.go
+++ b/s3_exporter.go
@@ -2,6 +2,7 @@ package main
 
 import (
        "net/http"
+       "os"
        "time"
 
        "github.com/prometheus/client_golang/prometheus"
@@ -142,16 +143,17 @@ func init() {
 
 func main() {
        var (
-               listenAddress = kingpin.Flag("web.listen-address", "Address to listen on for web interface and telemetry.").Default(":9340").String()
-               metricsPath   = kingpin.Flag("web.metrics-path", "Path under which to expose metrics").Default("/metrics").String()
-               probePath     = kingpin.Flag("web.probe-path", "Path under which to expose the probe endpoint").Default("/probe").String()
-               endpointUrl   = kingpin.Flag("s3.endpoint-url", "Custom endpoint URL for S3 service").Default("").Envar("S3_ENDPOINT_URL").String()
+               app           = kingpin.New(namespace+"_exporter", "Export metrics for S3 certificates").DefaultEnvars()
+               listenAddress = app.Flag("web.listen-address", "Address to listen on for web interface and telemetry.").Default(":9340").String()
+               metricsPath   = app.Flag("web.metrics-path", "Path under which to expose metrics").Default("/metrics").String()
+               probePath     = app.Flag("web.probe-path", "Path under which to expose the probe endpoint").Default("/probe").String()
+               endpointUrl   = app.Flag("s3.endpoint-url", "Custom endpoint URL for S3 service").Default("").String()
        )
 
        log.AddFlags(kingpin.CommandLine)
-       kingpin.Version(version.Print(namespace + "_exporter"))
-       kingpin.HelpFlag.Short('h')
-       kingpin.Parse()
+       app.Version(version.Print(namespace + "_exporter"))
+       app.HelpFlag.Short('h')
+       kingpin.MustParse(app.Parse(os.Args[1:]))
 
        var sess *session.Session
        var err error
```
This would make the environment variable `S3_EXPORTER_S3_ENDPOINT_URL`, which is a bit verbose but it makes it clearer that it belongs to the exporter and also follows the kingpin convention which is well understood by go users.

What do you think?

File: s3_exporter.go (hunk 1 above)
Comment (stroitelev) [reply s3_exporter.go]: Hello, sure, I have updated.

**Comment by stroitelev** [comment]
About the environment variable S3_EXPORTER_S3_ENDPOINT_URL, I think it is well.

**Comment by ribbybibby** [comment]
Hi @stroitelev, your fork doesn't seem to use any environment variables now. You'll need to make a few more changes to enable kingpin's default envvars (see the example from my previous comment).

Also, the naming of ` endpointURL` is inconsistently cased:
```
# github.com/stroitelev/s3_exporter
./s3_exporter.go:166:10: undefined: endpointUrl
./s3_exporter.go:167:21: undefined: endpointUrl
```

**Comment by ribbybibby** [comment]
Closing. I found some time to implement this myself. I've released it as 0.3.0.

**Pull request closed without merging**